### PR TITLE
Cleanup semantic segmentation metrics

### PR DESCRIPTION
**Pull request by guarin**
## What has changed and why?

* Only log important metrics by default
* Do not log metrics for all layers
* Adjusted validation metrics to reflect actual inference performance
* Make metric names clearer
* Make classwise metrics show class name

## How has it been tested?

* Manually

Screenshot 2025-08-01 at 13 50 15
Screenshot 2025-08-01 at 13 53 18

## Did you update [CHANGELOG.md](../CHANGELOG.md)?

- [ ] Yes
- [x] Not needed (internal change)

## Did you update the documentation?

- [ ] Yes
- [x] Not needed (internal change without effects for user)

### CODE REVIEW

File: src/lightly_train/_commands/train_task.py [hunk 1]
```diff
@@ -260,15 +260,10 @@ def _train_task_from_config(config: TrainTaskConfig) -> None:
...
-        )
-        is_save_ckpt_step = (
-            step > 0
-            and (step + 1) % no_auto(config.save_checkpoint_args.save_every_num_steps)
-            == 0
-        )
+        is_val_step = (step + 1) % no_auto(config.logger_args.val_every_num_steps) == 0
+        is_save_ckpt_step = (step + 1) % no_auto(
+            config.save_checkpoint_args.save_every_num_steps
+        ) == 0
```
Comment (guarin) [inline src/lightly_train/_commands/train_task.py:266]: Removed the `step > 0` condition, otherwise validation / checkpoint saving cannot run after the first step.

File: src/lightly_train/_commands/train_task.py [hunk 2]
```diff
@@ -286,13 +281,15 @@ def _train_task_from_config(config: TrainTaskConfig) -> None:
...
             for group in optimizer.param_groups:
-                train_log_dict[f"lr/{group['name']}"] = group["lr"]
-                train_log_dict[f"wd/{group['name']}"] = group["weight_decay"]
+                train_log_dict[f"learning_rate/{group['name']}"] = group["lr"]
+                train_log_dict[f"weight_decay/{group['name']}"] = group["weight_decay"]
             fabric.log_dict(train_log_dict, step=step)
             helpers.reset_metrics(train_result.log_dict)
 
+        if is_save_ckpt_step or is_last_step:
+            helpers.save_checkpoint(fabric=fabric, out_dir=out_dir, state=state)
```
Comment (guarin) [inline src/lightly_train/_commands/train_task.py:291]: Moved this further up to save the checkpoint before validation in case there is an error during validation.

File: src/lightly_train/_task_models/dinov2_eomt_semantic_segmentation/task_model.py [hunk 3]
```diff
@@ -143,7 +143,6 @@ def __init__(
         # TODO(Guarin, 07/25): Move all attention mask handling to the train module.
         # Attention mask prob can be passed as argument to forward_train. No need to
         # store it as a parameter here.
-        self.masked_attn_enabled = True
```
Comment (guarin) [inline src/lightly_train/_task_models/dinov2_eomt_semantic_segmentation/task_model.py:146]: Not needed anymore

File: src/lightly_train/_task_models/dinov2_eomt_semantic_segmentation/task_model.py [hunk 4]
```diff
@@ -211,16 +210,19 @@ def forward(self, x: Tensor) -> tuple[Tensor, Tensor]:
...
+        _, _, H, W = x.shape
         patch_size = self.backbone.patch_size
         grid_size = (H // patch_size, W // patch_size)
 
         x = self.backbone.prepare_tokens_with_masks(x)  # type: ignore[no-untyped-call]
-        attn_mask = None
         mask_logits_per_layer, class_logits_per_layer = [], []
 
         for i, block in enumerate(self.backbone.blocks):
+            attn_mask = None
```
Comment (guarin) [inline src/lightly_train/_task_models/dinov2_eomt_semantic_segmentation/task_model.py:224]: Moved this inside the loop to make it more evident that a new attention mask is created in each iteration.

File: src/lightly_train/_task_models/dinov2_eomt_semantic_segmentation/train_model.py [hunk 5]
```diff
@@ -140,10 +140,12 @@ def __init__(
...
         )
+
+        # Metrics
         self.val_loss = MeanMetric()
-        # MeanIoU assumes that background is class 0.
-        # TODO(Guarin, 07/25): Make params configurable.
-        # TODO(Thomas, 07/25): Use self.train_metrics only.
+
+        # TODO(Guarin, 08/25): Speed up metric calculation by not calculating
+        # mIoU and classwise IoU separately. mIoU can be derived from the classwise IoU.
```
Comment (stegmuel) [inline src/lightly_train/_task_models/dinov2_eomt_semantic_segmentation/train_model.py:148]: This confuses me. I still see two calls to two difference metrics.

**Review by stegmuel — COMMENTED** [review]
Left a question.

File: src/lightly_train/_task_models/dinov2_eomt_semantic_segmentation/train_model.py (hunk 5 above)
Comment (guarin) [reply src/lightly_train/_task_models/dinov2_eomt_semantic_segmentation/train_model.py:148]: It is a todo
